components/encryption/src/manager/mod.rs

```diff
@@ -591,7 +593,12 @@ impl DataKeyManager {
             writer,
             crypter::encryption_method_from_db_encryption_method(file.method),
             &file.key,
-            Iv::from_slice(&file.iv)?,
+            if file.iv.is_empty() {
```

Comment:
I'm concerned about adding this logic will hide bugs that forget to register file name and thus the target file will be not encrypted when encryption is enabled. Are there any case we actually need this fallback logic?

Reply:
That's a fair point. But right now RocksDB also allows opening plaintext file:
https://github.com/tikv/rocksdb/blob/21712fe4243a57a5e7854574b933e67d0fd227f6/encryption/encryption.cc#L262. The two should be consistent.

With this patch, we should be able to enable encryption on an un-encrypted dataset. I honestly don't like using encryption this way. But QA thinks it's useful for their testing.